catalog/transformer/catalog-transformer-csv-queryresponse/src/main/java/ddf/catalog/transformer/csv/CsvQueryResponseTransformer.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.transformer.csv;
+
+import static ddf.catalog.transformer.csv.common.CsvTransformer.createResponse;
+import static ddf.catalog.transformer.csv.common.CsvTransformer.getAllCsvAttributeDescriptors;
+import static ddf.catalog.transformer.csv.common.CsvTransformer.sortAttributes;
+import static ddf.catalog.transformer.csv.common.CsvTransformer.writeMetacardsToCsv;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.QueryResponseTransformer;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * An implementation of QueryResponseTransformer that produces CSV output.
+ *
+ * @see ddf.catalog.transform.QueryResponseTransformer
+ */
+public class CsvQueryResponseTransformer implements QueryResponseTransformer {
+
+  private static final String COLUMN_ORDER_KEY = "columnOrder";
+
+  private static final String COLUMN_ALIAS_KEY = "aliases";
+
+  /**
+   * @param upstreamResponse the SourceResponse to be converted.
+   * @param arguments this transformer accepts 2 parameters in the 'arguments' map.
+   *     <ol>
+   *       <li>key: 'attributeOrder' value: a java.utilList containing Attribute name (as Strings)
+   *           to identify the order that the columns will appear in the output.
+   *       <li>key: 'aliases' value: a java.util.Map whose keys are attribute names and values are
+   *           the how that attribute column should be aliased in the output. For example, if the
```

Comment:
How about "...and values are the corresponding column headers that will be printed in the output."?